numpy/lib/npyio.py

```diff
@@ -791,7 +792,14 @@ def loadtxt(fname, dtype=float, comments='#', delimiter=None,
     """
     # Type conversions for Py3 convenience
     if comments is not None:
-        comments = asbytes(comments)
+        if isinstance(comments, (np.compat.basestring, np.compat.bytes)):
```

Comment:
You mean I should check against `basestring` only ? It could potentially create a compatibility issue.